lib/remote/eventshandler.cpp

```diff
@@ -105,7 +106,7 @@ bool EventsHandler::HandleRequest(
 	response.result(http::status::ok);
 	response.set(http::field::content_type, "application/json");
 
-	IoBoundWorkSlot dontLockTheIoThread (yc);
+	handlingRequest.Done();
```

Comment:
I've figured out and pushed how to get rid of this and the 38 changed files.

```
$ git diff --stat 74009f0fc..a663f98b2
 lib/base/io-engine.cpp              | 77 +++++++++++++++++++++++++++++++++++++++--------------------------------------
 lib/base/io-engine.hpp              | 64 +++++++++++++++++++++++++++++++++-------------------------------
 lib/remote/eventshandler.cpp        |  2 --
 lib/remote/httpserverconnection.cpp | 19 +++++++++++++------
 lib/remote/httpserverconnection.hpp |  2 ++
 lib/remote/jsonrpcconnection.cpp    |  2 +-
 6 files changed, 88 insertions(+), 78 deletions(-)
$
```

Please let me know whether you consider this better.

Reply:
CC @julianbrost @yhabteab @oxzi